repository/src/main/java/org/apache/atlas/tasks/TaskRegistry.java

```diff
@@ -245,14 +245,15 @@ public void deleteComplete(AtlasVertex taskVertex, AtlasTask task) {
         deleteVertex(taskVertex);
     }
 
-    public void inProgress(AtlasVertex taskVertex, AtlasTask task) {
+    public void inProgress(AtlasVertex taskVertex, AtlasTask task, AbstractTask runnableTask) {
         RequestContext.get().setCurrentTask(task);
 
         task.setStartTime(new Date());
-
+        task.setAssetsCountToPropagate(runnableTask.getAssetsCountToPropagate());
         setEncodedProperty(taskVertex, Constants.TASK_START_TIME, task.getStartTime());
         setEncodedProperty(taskVertex, Constants.TASK_STATUS, AtlasTask.Status.IN_PROGRESS);
         setEncodedProperty(taskVertex, Constants.TASK_UPDATED_TIME, System.currentTimeMillis());
+        setEncodedProperty(taskVertex, Constants.TASK_ASSET_COUNT_TO_PROPAGATE, task.getAssetsCountToPropagate());
```

Comment:
Are you sure you have the real number here to set assetsCountToPropagate?
IMO here you will always get it as 0, as per previous code I saw while creating Task vertex we already write value as 0, I feel this is not needed